rkt/image/httpops.go

```diff
@@ -89,17 +89,21 @@ func (o *httpOps) DownloadImageWithETag(u *url.URL, etag string) (readSeekCloser
 	ensureLogger(o.Debug)
 	aciFile, err := getTmpROC(o.S, u.String())
 	if err != nil {
-		return nil, nil, err
+		return NopReadSeekCloser(nil), nil, err
 	}
-	defer func() { maybeClose(aciFile) }()
+	defer func() {
+		if aciFile != nil {
+			aciFile.Close()
+		}
+	}()
```

Comment:
Don't close the `aciFile` here, since it is being returned, hence it is the caller's responsibility to close it. I'd remove these lines alltogether since `getTmpROC` now maintains the invariant of a nil-safe readSeekCloser.